slightly smarter starter

JohnCardinal opened the pull request:
Players give clues about 1s to the next player. If they know they have a 1 they will play it instead of playing the first card in their hand

JohnCardinal committed:
Merge pull request #1 from Spurlock/master

More games. Predictable scoring.

JohnCardinal committed:
Slightly smarter starter

moved main loop into player.do_somthing. Player will now give clues to
the next player and plays cards that they know are 1s before defaulting
to playing their first card. Added print and exit() when player gives
themself a clue. Also moved print knowledge to give clue for
receiving_player and added print "player turn" at the start of main
loop.

Spurlock commented:
Thanks John!

I might have miscommunicated what I had in mind for do_something(). Basically I think that method should be the part where the current player decides what to do with his turn. So we don't want to move the entire main loop up into there, only the part where the player decides what to do and then does it. 

take_turn() might be a better name for that method.

JohnCardinal committed:
take_turns organization

changed do_something to take_turn and moved game over and last turn
checks back near game start

JohnCardinal committed:
Player numbers

Added player.seats to issue a number to each player. Also added a bit of
print text to help read the turn actions of when run

JohnCardinal commented:
I reorganized the turn loop to only include turn stuff and gave each player a set number in the player _init_. I also added a bit more print text to help me read the turn actions after running. I started to try and find a way to check every player but the current one, but that got a bit more complicated, so I just stuck with the next player after the current. I'm also not sure if I need to send a new pull request or if my new commits can piggyback on this one. if not the latter or if it needs more tweaks lemme know.
